apps/console/src/features/applications/utils/sign-in-method-utils.ts

```diff
@@ -396,9 +396,51 @@ export class SignInMethodUtils {
                 idpList: []
             };
         }
+    }
+
+    /**
+     * Returns the count of 2FAs in a certain authentication step.
+     *
+     * @param currentStep - The current step.
+     * @param authenticationSteps - The authentication steps.
+     *
+     * @returns number of 2FAs
+     */
+    public static countTwoFactorAuthenticatorsInCurrentStep(
+        currentStep: number, authenticationSteps: AuthenticationStepInterface[]
+    ): number {
+        const secondFactor: AuthenticatorInterface[] = authenticationSteps[currentStep].options.filter(
+            (authenticator: AuthenticatorInterface) =>
+                authenticator.authenticator === IdentityProviderManagementConstants.TOTP_AUTHENTICATOR ||
+                authenticator.authenticator === IdentityProviderManagementConstants.EMAIL_OTP_AUTHENTICATOR ||
+                authenticator.authenticator === IdentityProviderManagementConstants.SMS_OTP_AUTHENTICATOR
+        );
 
+        return secondFactor.length;
     }
 
+    /**
+     * Checks whether a specified authenticator is there in the current step.
+     *
+     * @param currentStep - The current step.
+     * @param authenticationSteps - The authentication steps.
+     *
+     * @returns boolean
+     */
+    public static hasSpecificAuthenticatorInCurrentStep(
+        authenticator: string, currentStep: number, authenticationSteps: AuthenticationStepInterface[]
+    ): boolean {
+        const hasAuthenticator: AuthenticatorInterface = authenticationSteps[currentStep].options.find(
+            (el: AuthenticatorInterface) =>
+                el.authenticator === authenticator
+        );
+
+        if (hasAuthenticator) {
+            return true;
+        } else {
+            return false;
+        }
```

Comment:
We can make this also less verbose, can't we?

```suggestion
        return hasAuthenticator
```